compiler/ballerina-lang/src/main/java/io/ballerina/projects/PackageCompilation.java

```diff
@@ -209,8 +210,12 @@ private void compileModulesInternal() {
             for (ModuleContext moduleContext : packageResolution.topologicallySortedModuleList()) {
                 moduleContext.compile(compilerContext);
                 for (Diagnostic diagnostic : moduleContext.diagnostics()) {
-                    diagnostics.add(new PackageDiagnostic(diagnostic, moduleContext.descriptor(),
-                            moduleContext.project()));
+                    if (!ProjectKind.BALA_PROJECT.equals(moduleContext.project().kind()) ||
```

Comment:
The condition to check if the build option is set is missing here.

Reply:
This is required for the LS. As we discussed offline, we don't pass it here because build options/ compilation options can't be passed up to this point.